### PR TITLE
[REF] spreadsheet: rename pivot functions

### PR DESCRIPTION
**Pull request by pro-odoo**
In order to prepare the introduction of a new kind of pivot (spreadsheet),
this commit renames the pivot functions:
- ODOO.PIVOT -> PIVOT.VALUE
- ODOO.PIVOT.HEADER -> PIVOT.HEADER
- ODOO.PIVOT.TABLE -> PIVOT

Task: 3748561

### CODE REVIEW

File: addons/spreadsheet/static/src/helpers/model.js [hunk 1]
```diff
@@ -163,7 +164,9 @@ function containsOdooFunction(content) {
     if (
         !content ||
         !content.startsWith("=") ||
-        (!content.toUpperCase().includes("ODOO.") && !content.toUpperCase().includes("_T"))
+        (!content.toUpperCase().includes("ODOO.") &&
+            !content.toUpperCase().includes("_T") &&
+            !content.toUpperCase().includes("PIVOT"))
```
Comment (hokolomopo) [inline addons/spreadsheet/static/src/helpers/model.js:169]: Function name is confusing now, `containsOdooFunction` returns true if there is a non-Odoo pivot function in the content. Maybe the check thatthe pivot type is Odoo should be done here ?

File: addons/spreadsheet/static/src/o_spreadsheet/migration.js [hunk 2]
```diff
@@ -16,6 +16,12 @@ const MAP = {
     "LIST.HEADER": "ODOO.LIST.HEADER",
 };
 
+const MAP_V10 = {
```
Comment (hokolomopo) [inline addons/spreadsheet/static/src/o_spreadsheet/migration.js:19]:
```suggestion
const MAP_FN_NAMES_V10 = {
```
or something like that, to be more explicit than just `MAP` ?

**Review by hokolomopo — COMMENTED** [review]
Didn't read each changed line, I'm gonna trust your Regex :stuck_out_tongue:

File: addons/spreadsheet/static/src/helpers/model.js (hunk 1 above)
Comment (pro-odoo) [reply addons/spreadsheet/static/src/helpers/model.js:169]: If I do the check here, I have to give this function the model, the sheet and the xc and change the code to make something way longer:
```js
return iterateAstNodes(ast).some((ast) => {
    if (ast.type !== "FUNCALL") {
        return false;
    }
    if (
        ast.value.toUpperCase().startsWith("ODOO.") ||
        ast.value.toUpperCase().startsWith("_T")
    ) {
        return true;
    }
    if (ast.value.toUpperCase().startsWith("PIVOT")) {
        // check if the pivot is an odoo pivot
        const { col, row } = toCartesian(xc);
        const position = { sheetId, col, row };
        const pivotId = model.getters.getPivotIdFromPosition(position);
        return pivotId && model.getters.getPivotDefinition(pivotId).type == "ODOO";
    }
});
```

Is it ok for you to rename this function (only used in this file) to `containsOdooCandidateFunctions` or `containsPossibleLoadedFunctions`, or something like that ?

**Pull request closed without merging**